catalog/nato/stanag4559/nato-stanag4559-source/src/main/java/org/codice/alliance/nato/stanag4559/source/Stanag4559FilterFactory.java

```diff
@@ -0,0 +1,301 @@
+
+    private Map<String, List<AttributeInformation>> queryableAttributes;
+
+    private String view;
+
+    public Stanag4559FilterFactory(Map<String, List<AttributeInformation>> queryableAttributes,
+            String view) {
+        this.queryableAttributes = queryableAttributes;
+        this.view = view;
+    }
+
+    public String buildPropertyIsLike(String value) {
+        if (queryableAttributes == null) {
+            return Stanag4559FilterDelegate.EMPTY_STRING;
+        }
+
+        List<AttributeInformation> attributeInformationList = queryableAttributes.get(view);
+
+        if (attributeInformationList == null) {
+            return Stanag4559FilterDelegate.EMPTY_STRING;
+        }
+
+        List<String> filters = new ArrayList<>();
+
+        // Replace * with %, since * is not a valid wildcard in BQS
+        value = value.replaceAll("\\*", "%");
+        value = value.replaceAll("\\?", "%");
+
+        for (AttributeInformation attributeInformation : attributeInformationList) {
+            if (isTextAttributeType(attributeInformation)) {
+                filters.add(LP + attributeInformation.attribute_name + LIKE
+                        + Stanag4559FilterDelegate.SQ + value + Stanag4559FilterDelegate.SQ + RP);
+            }
+        }
+        return buildOrFilter(filters);
+    }
+
+    public boolean isTextAttributeType(AttributeInformation attributeInformation) {
+        return attributeInformation.attribute_type.equals(AttributeType.TEXT)
+                && !attributeInformation.attribute_name.equals(TYPE);
+    }
+
+    public String buildPropertyIsEqualTo(String property, String value) {
+        property = mapToNsil(property);
+        return LP + property + EQ + value + RP;
+    }
+
+    public String buildPropertyIsNotEqualTo(String property, String value) {
+        property = mapToNsil(property);
+        return buildNotFilter(buildPropertyIsEqualTo(property, value));
+    }
+
+    public String buildPropertyIsGreaterThan(String property, String value) {
+        property = mapToNsil(property);
+        return LP + property + GT + value + RP;
+    }
+
+    public String buildPropertyIsGreaterThanOrEqual(String property, String value) {
+        property = mapToNsil(property);
+        return LP + property + GTE + value + RP;
+    }
+
+    public String buildPropertyIsLessThan(String property, String value) {
+        property = mapToNsil(property);
+        return LP + property + LT + value + RP;
+    }
+
+    public String buildPropertyIsLessThanOrEqual(String property, String value) {
+        property = mapToNsil(property);
+        return LP + property + LTE + value + RP;
+    }
+
+    public String buildPropertyIsBetween(String property, String lowerBound, String upperBound) {
+        property = mapToNsil(property);
+        return LP + property + BTW + lowerBound + COMMA + upperBound + RP;
+    }
+
+    public String buildOrFilter(List<String> filters) {
+        if (filters.size() == 0) {
```

Comment:
You could delete this if you wanted to since you're returning `Stanag4559FilterDelegate.EMPTY_STRING` at the end if the `StringBuilder` doesn't have any filters in it.